Src/AutoNSubstituteUnitTest/SubstituteBuilderTest.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using NSubstitute;
+using Ploeh.AutoFixture.Kernel;
+using Xunit;
+
+namespace Ploeh.AutoFixture.AutoNSubstitute.UnitTest
+{
+    public class SubstituteBuilderTest
+    {
+        [Fact]
+        public void ClassImplementsISpecimenBuilderToServeAsFixtureCustomization()
+        {
+            // Fixture setup
+            // Exercise system
+            // Verify outcome
+            Assert.True(typeof(ISpecimenBuilder).IsAssignableFrom(typeof(SubstituteBuilder)));
+            // Teardown
+        }
+
+        [Fact]
+        public void ConstructorThrowsArgumentNullExceptionWhenSubstituteConstructorIsNull()
+        {
+            // Fixture setup
+            // Exercise system
+            var e = Assert.Throws<ArgumentNullException>(() => new SubstituteBuilder((ISpecimenBuilder)null));
+            // Verify outcome
+            Assert.Equal("substituteConstructor", e.ParamName);
+            // Teardown
+        }
+
+        [Fact]
+        public void SubstituteConstructorReturnsValueSpecifiedInConstructorToEnableTestingOfCustomizations()
+        {
+            // Fixture setup
+            var substituteConstructor = Substitute.For<ISpecimenBuilder>();
+            // Exercise system
+            var sut = new SubstituteBuilder(substituteConstructor);
+            // Verify outcome
+            Assert.Same(substituteConstructor, sut.SubstituteConstructor);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsNoSpecimenWhenRequestIsNotAnExplicitSubstituteRequest()
+        {
+            // Fixture setup
+            var constructor = Substitute.For<ISpecimenBuilder>();
+            var sut = new SubstituteBuilder(constructor);
+            var request = typeof(IComparable);
+            var context = Substitute.For<ISpecimenContext>();
+            // Exercise system
+            object result = sut.Create(request, context);
+            // Verify outcome
+            var noSpecimen = Assert.IsType<NoSpecimen>(result);
+            Assert.Same(request, noSpecimen.Request);

```

Comment:
`NoSpecimen` has structural equality, so it'd be easier to read the assertion if it was expressed like this:

```C#
var expected = new NoSpecimen(request);
Assert.Equal(expected, specimen);
```